src/VectorBundle.jl

```diff
@@ -0,0 +1,461 @@
+
+"""
+    VectorBundle{𝔽,TVS,TM,TVT}
+
+Alias for [`FiberBundle`](@ref) when fiber type is a `TVS` of type
+[`VectorSpaceType`](https://juliamanifolds.github.io/ManifoldsBase.jl/stable/bases.html#ManifoldsBase.VectorSpaceType).
+
+`VectorSpaceFiberType` is used to encode vector spaces as fiber types.
+"""
+const VectorBundle{𝔽,TVS,TM,TVT} = FiberBundle{
+    𝔽,
+    VectorSpaceFiberType{TVS},
+    TM,
+    TVT,
+} where {
+    TVS<:VectorSpaceType,
+    TM<:AbstractManifold{𝔽},
+    TVT<:FiberBundleProductVectorTransport,
+}
+
+function VectorBundle(
+    vst::VectorSpaceType,
+    M::AbstractManifold,
+    vtm::FiberBundleProductVectorTransport,
+)
+    return FiberBundle(VectorSpaceFiberType(vst), M, vtm)
+end
+function VectorBundle(vst::VectorSpaceType, M::AbstractManifold)
+    return FiberBundle(VectorSpaceFiberType(vst), M)
+end
+
+"""
+    TangentBundle{𝔽,M} = VectorBundle{𝔽,TangentSpaceType,M} where {𝔽,M<:AbstractManifold{𝔽}}
+
+Tangent bundle for manifold of type `M`, as a manifold with the Sasaki metric [Sasaki:1958](@cite).
+
+Exact retraction and inverse retraction can be approximated using [`FiberBundleProductRetraction`](@ref),
+[`FiberBundleInverseProductRetraction`](@ref) and [`SasakiRetraction`](@ref).
+[`FiberBundleProductVectorTransport`](@ref) can be used as a vector transport.
+
+# Constructors
+
+    TangentBundle(M::AbstractManifold)
+    TangentBundle(M::AbstractManifold, vtm::FiberBundleProductVectorTransport)
+"""
+const TangentBundle{𝔽,M} =
+    VectorBundle{𝔽,TangentSpaceType,M} where {𝔽,M<:AbstractManifold{𝔽}}
+
+TangentBundle(M::AbstractManifold) = VectorBundle(TangentSpace, M)
+function TangentBundle(M::AbstractManifold, vtm::FiberBundleProductVectorTransport)
+    return VectorBundle(TangentSpace, M, vtm)
+end
+
+function default_inverse_retraction_method(::TangentBundle)
+    return FiberBundleInverseProductRetraction()
+end
+
+function default_retraction_method(::TangentBundle)
+    return FiberBundleProductRetraction()
+end
+
+function default_vector_transport_method(B::TangentBundle)
+    default_vt_m = default_vector_transport_method(B.manifold)
+    return FiberBundleProductVectorTransport(default_vt_m, default_vt_m)
+end
+
+"""
+    distance(B::BundleFibers, p, X, Y)
+
```

Comment:
While maybe slightly outdates (the function defines this for VectorBundleFibers) now after vector bundle being a special case of fiber bundles – I am actually finally lost on Vector bundle fibers What are these?

Reply:
On trivial fiber bundles (and basically all fiber bundles we want to work on are trivial) operations generally decompose into manifold-part and fiber-part. `BundleFibers` is used to the fiber-part computations. It's a bit of abuse of notation actually and I'm not sure it's the right structure but it works well enough so far. So, here we don't consider `BundleFibers` as a manifold exactly, but the fiber attached to point `p` in the fiber bundle, and `X` and `Y` are fiber-projections of a point on the fiber bundle.